src/jdk.incubator.vector/share/classes/jdk/incubator/vector/AbstractVector.java

```diff
@@ -736,6 +742,25 @@ AbstractVector<F> convert0(char kind, AbstractSpecies<F> rsp) {
         throw new AssertionError();
     }
 
+    /*package-private*/
+    @ForceInline
+    final
+    AbstractMemorySegmentImpl requireSegmentConvertibleFor(MemorySegment segment, long offset, int elementByteSize)  {
+        AbstractMemorySegmentImpl ams = (AbstractMemorySegmentImpl) segment;
+        if (ams.maxAlignMask() > 1 && !ams.isAlignedForElement(offset, elementByteSize)) {
```

Comment:
I don't think we should only check for maxAlignMask > 1 - you also want to check native segments, right?